File: backend/src/main/java/com/simats/neoomfs/dto/response/UserProfileResponse.java
package com.simats.neoomfs.dto.response;

import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

/**
 * User profile response DTO.
 */
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserProfileResponse {
    private Long id;
    private String fullName;
    private String username;
    private String email;
    private String licenseNumber;
    private String department;
    private String institution;
    private String phoneNumber;
    private boolean active;
    private List<String> roles;
    private LocalDateTime lastLogin;
    private LocalDateTime createdAt;
}

File: frontend/app/src/main/java/com/simats/neoomfs/models/UserProfileResponse.kt
package com.simats.neoomfs.models

data class UserProfileResponse(
    val id: Long,
    val fullName: String,
    val username: String,
    val email: String,
    val licenseNumber: String?,
    val department: String?,
    val institution: String?,
    val phoneNumber: String?,
    val active: Boolean,
    val roles: List<String>,
    val lastLogin: String?,
    val createdAt: String?
)
